src/pyenphase/updaters/generator.py

```diff
@@ -0,0 +1,50 @@
+import logging
+from typing import Any
+
+from ..const import (
+    ENSEMBLE_MIN_VERSION,
+    URL_GEN_CONFIG,
+    SupportedFeatures,
+)
+from ..exceptions import ENDPOINT_PROBE_EXCEPTIONS
+from ..models.envoy import EnvoyData
+from .base import EnvoyUpdater
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class EnvoyGeneratorUpdater(EnvoyUpdater):
+    """Class to handle updates for Generator information."""
+
+    async def probe(
+        self, discovered_features: SupportedFeatures
+    ) -> SupportedFeatures | None:
+        """Probe the Envoy for this updater and return SupportedFeatures."""
+        if self._envoy_version < ENSEMBLE_MIN_VERSION:
+            _LOGGER.debug("Firmware too old for Ensemble support")
+            return None
+
+        # Check for generator support
+        try:
+            result = await self._json_probe_request(URL_GEN_CONFIG)
+        except ENDPOINT_PROBE_EXCEPTIONS as e:
```

Comment:
We are starting to do quite a lot of probes at startup.

I'm wondering if there is some data we already have that tells us if there is a generator or not that would allow us to skip the probe (and maybe some others)

Reply:
We may be able to check for SupportedFeatures.ENPOWER as that's a dependency for generators as well, but an Enpower does not guarantee a generator

Reply:
Maybe /inventory or /inv shows them all? For my simple envoy it shows the envoy, al inverters and the Q relay.

Reply:
Does the probe system guarantee order or that previous probes will complete before starting a new one? If there's no guarantee of order or if it will complete then we probably can't check for the Enpower supported feature

Reply:
Yes it's ordered and logic is depending on it . This allows updaters to have fallbacks if needed.